packages/dataviews/src/dataviews-layouts/list/index.tsx

```diff
@@ -175,6 +175,8 @@ function ListItem< Item >( {
 		};
 	}, [ actions, item ] );
 
+	const hasOnlyOnePrimaryAction = primaryAction && actions.length === 1;
```

Comment:
Note this checks `actions`, not `eligibleActions`. This is intentional to make sure that we only hide the item menu when *all* the items has a single action. We wouldn't want to show only the icon for some and the icon + the actions menu for others.